Octokit/Models/Response/RepositoryPermissions.cs

```diff
@@ -20,19 +22,32 @@ public RepositoryPermissions(bool admin, bool push, bool pull)
         /// </summary>
         public bool Admin { get; protected set; }
 
+        /// <summary>
+        /// Whether the current user has maintain permissions
+        /// </summary>
+        public bool Maintain { get; protected set; }
```

Comment:
**Non-Blocker**: 
Would you mind changing all of these property accessors (on the `set`) to the more appropriate (given the use case) `private`? Using `protected` is currently the dominant pattern in this SDK but given that most of our models now implement `ctors` in one form or another so that makes it where I think we could change most of the protected access modifiers to the more appropriate private ones.

Let me know your thoughts - I'll be making a push to change all of the model's properties to use the private modifier soon so I am good with whatever you decide. Once done I'll be adding a theory to our [model tests](https://github.com/octokit/octokit.net/blob/main/Octokit.Tests.Conventions/ModelTests.cs) to help enforce this.

Reply:
Sure thing, I wasn't sure whether the JSON deserializer in use supported private props.
I made them get-only in 40bdd8535ed6ef8a9303b9bf40b1533f6781ba40 or is that pushing the limits :D

Reply:
✅ I could go for that... given C# 6  and greater these two are essentially equivalent when it comes to constructors but deviate when it comes to methods internal to the model needing to mutate the properties. I really don’t think there will ever be methods that mutate props on response models; that feels like a code smell - but if there is ever a need we can introduce private again. 

I think I like your omitted setter idea better than private for future generated response models. 🎉 Thanks for the thought and effort you put in here!  We should be good to merge this.

Reply:
I'm afraid I've been a bit too zealous.
While debugging this branch in my production code the ctor with the 5 arguments is never hit?

Reply:
Hey @janv8000, so I dug into this a bit more.  And it looks like you're correct.  We need to at minimum implement a setter with the private modifier.

`SimpleJson` makes heavy use of reflection to:
1. Get the [constructor](https://github.com/octokit/octokit.net/blob/main/Octokit/SimpleJson.cs#L1945) and a [delegate](https://github.com/octokit/octokit.net/blob/main/Octokit/SimpleJson.cs#L1958) for the constructor
2. Iterate through a dictionary (created from the incoming payload) and [assign values](https://github.com/octokit/octokit.net/blob/main/Octokit/SimpleJson.cs#L1495) to each given setter

This is why you only see the default constructor getting hit and not the parameter-based one on incoming payloads.  

 I am making an assumption here, but I believe that the parameterized constructor is primarily used for testing/mocking - because the delegate uses the memoized object map to make the assignments.

Let me know if that makes sense, and apologies again for missing your question the first time around.